core/src/main/scala/com/bcf/statefun4s/StatefulFunction.scala

```diff
@@ -71,6 +71,7 @@ trait StatefulFunction[F[_], S] {
   def modifyCtx(modify: S => S): F[Unit]
   def deleteCtx: F[Unit]
   def myAddr: F[Address]
+  def caller: F[Address]
```

Comment:
instead of raising, could this be `F[Option[Address]]`?

Reply:
My thought was that all the error handling around caller not being there isn't worth it given than pretty much only 1 function could have no caller anyways.

Reply:
I think I'll add `callerOption` to be sort of like `head` and `headOption`